cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -342,6 +344,27 @@ public void deleteAllApps(HttpRequest request, HttpResponder responder,
     responder.sendStatus(HttpResponseStatus.OK);
   }
 
+  private ArtifactScope validateScope(String scope) throws BadRequestException {
+    try {
+      return ArtifactScope.valueOf(scope.toUpperCase());
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException("Invalid scope " + scope);
+    }
+  }
+
+  private Set<ArtifactScope> getArtifactScopesToConsider(Set<String> artifactScopes) throws BadRequestException {
+    Set<ArtifactScope> artifactScopesToConsider = new HashSet<>();
```

Comment:
Can be simplified as:

```java
if (artifactScopes.isEmpty()) {
  return EnumSet.allOf(ArtifactScope.class);
}
return artifactScopes.stream().map(ArtifactScope::valueOf).collect(Collectors.toSet());
```

Reply:
Could not use lambda cleanly cause I wanted to use validateScope which throws an exception.